src/pages/PasswordReset/index.tsx

```diff
@@ -0,0 +1,161 @@
+import { type SubmitHandler, useForm } from 'react-hook-form';
+import { type PasswordUpdateData } from '../../types';
+
+const SetNewPasswordDataSchema = z.object({
+  newPassword: z.string(),
+  token: z.string(),
+});
+
+const NewPasswordModal: React.FC = () => {
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [message, setMessage] = useState('');
+  const { updatePassword } = useResetPassword();
+  const [loading, setLoading] = useState(false);
+  const [isOpen, setIsOpen] = useState(true);
+  const { register, handleSubmit, setValue } = useForm<PasswordUpdateData>();
+  const [messageType, setMessageType] = useState('default');
+
+  const location = useLocation();
+  const navigate = useNavigate();
+  const token = new URLSearchParams(location.search).get('token') ?? '';
+
+  const handleConfirmPasswordChange = (event: {
+    target: { value: React.SetStateAction<string> };
+  }) => {
+    setConfirmPassword(event.target.value);
+  };
+  useEffect(() => {
+    setValue('token', token);
+  }, [token, setValue]);
+
+  const onSubmit: SubmitHandler<PasswordUpdateData> = async (data) => {
+    setLoading(true);
+    const validatedData = SetNewPasswordDataSchema.parse(data);
+    if (validatedData.newPassword !== confirmPassword) {
+      setMessage('Passwords do not match. Please try again.');
+      setMessageType('error');
+      setLoading(false);
+      return;
+    }
+    setMessage('');
+    updatePassword(validatedData, {
+      onSuccess: () => {
+        setMessage('Your password has been reset successfully.');
+        setLoading(false);
+        setMessageType('success');
+      },
+      onError: () => {
+        setMessage('Error resetting password. Please try again.');
+        setMessageType('error');
+        setLoading(false);
+      },
+    });
+  };
+
+  const handleClose = (
+    _event: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ): void => {
+    setIsOpen(false);
+    navigate('/');
+  };
+
+  return (
+    isOpen && (
+      <div className="fixed z-10 inset-0 overflow-y-auto">
+        <div className="flex min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0 flex-col items-center justify-center h-screen w-full p-5 ">
+          <div className="fixed inset-0 transition-opacity">
+            <div className="absolute inset-0 bg-gray-500 opacity-75"></div>
+          </div>
+          <span className="hidden sm:inline-block sm:align-middle sm:h-screen"></span>
+          <div
+            className="inline-block align-bottom bg-white rounded-lg text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle sm:max-w-lg sm:w-full"
+            role="dialog"
+            aria-modal="true"
+            aria-labelledby="modal-headline"
+          >
+            <button
+              className="absolute top-2 right-2 p-2 text-gray-700 hover:text-gray-900 focus:outline-none"
+              onClick={handleClose}
+            >
```

Comment:
Use a <Link></Link> here. import it from react router

Reply:
so i need to navigate part right ?